### PR TITLE
Use `append_span` in forge_std + remove unnecessary imports

### LINKED ISSUE #1379: Use `append_span` in snforge_std
### Which component is your feature related to?

Forge

### Feature Request

👍

### PR DESCRIPTION
**Pull request by TAdev0**
Closes #1379 

## Introduced changes

This PR uses `append_span` where its is needed in forge_std, i.e. directly instead  of `extend_array` function in tx_info.cairo.

I also removed unnecessary imports, not needed with prelude `2023_10` used by default with `scarb >= 2.4.0`

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [x] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: snforge_std/src/cheatcodes/tx_info.cairo [hunk 1]
```diff
@@ -70,10 +65,5 @@ fn stop_spoof(target: CheatTarget) {
 }
 
 fn extend_array(ref array: Array<felt252>, mut span: Span<felt252>) {
-    loop {
-        match span.pop_front() {
-            Option::Some(x) => { array.append(x.clone()); },
-            Option::None => { break; }
-        };
-    };
+    array.append_span(span);
```
Comment (piotmag769) [inline snforge_std/src/cheatcodes/tx_info.cairo:68]: Can we remove `extend_array` altogether and just use this method instead?

**Review by piotmag769 — COMMENTED** [review]
LGTM, just one nit

File: snforge_std/src/cheatcodes/tx_info.cairo (hunk 1 above)
Comment (TAdev0) [reply snforge_std/src/cheatcodes/tx_info.cairo:68]: sure!

**Review by piotmag769 — APPROVED** [review]
Thanks a lot!

**Pull request merged**